cpp/src/arrow/type.h

```diff
@@ -449,7 +467,10 @@ class ARROW_EXPORT CTypeImpl : public BASE {
 
   int bit_width() const override { return static_cast<int>(sizeof(C_TYPE) * CHAR_BIT); }
 
-  DataTypeLayout layout() const override { return {{1, bit_width()}, false}; }
+  DataTypeLayout layout() const override {
+    return DataTypeLayout(
+        {DataTypeLayout::Bitmap(), DataTypeLayout::FixedWidth(sizeof(C_TYPE))});
```

Comment:
General question applicable to all types, when `nullable=false`, should we return `AlwaysNull` instead of `Bitmap` for the bitmap buffer?

Reply:
I don't think so. There could be a non-null all-ones bitmap buffer.